### DISCUSSION TITLE
Spatialized audio not working

### DISCUSSION THREAD

**Discussion by imjusthere** [microsoft/MixedRealityToolkit-Unity#11334] [Q&A] [answered]
I set the Spatializer Plugin to Microsoft Spatializer in Project Settings>Audio>Spatializer Plugin. I am building for the Quest 2 using OpenXR. If I check the Spatialize checkbox on any audio source I get a warning saying:

 "Audio source failed to initialize audio spatializer. Please check that the project audio settings and make sure that the selected spatializer is compatible with the target."

It looks like the demo scenes don't have Spatialized checked on any audio sources so I've turned that off. I also tried changing the spatializer to "OculusSpatializer" but that gave me the same warning as above if I checked "Spatialize" and didn't seem to have any effect with it off. What am I missing here?

**Comment by Zee2** [comment, marked as answer]
Hi! Which version of MRTK are you working with? Also, which version of the MS Spatializer?

**Reply by imjusthere** [reply]
It's MRTK3 (v1.0.2209.0) and I updated the spatializer to 2.0.37.

**Reply by Zee2** [reply]
@davidkline-ms can talk more, but Spatializer v2 is a pain in the butt to set up. The docs for the spatializer are also really bad at explaining what to do. You also need an audio mixer in your project with the spatialization channel set up with the spatializer filter.... super annoying. You can see the PR where we added our audio mixer for this here: https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11271

GH repo for Spatializer https://github.com/microsoft/spatialaudio-unity also doesn't have the right documentation either. I've made it clear to the audio/spatializer teams that this docs situation is terrible and needs to be fixed.

**Reply by Zee2** [reply]
Here's the chapter buried in the tutorial explaining the audio mixer https://learn.microsoft.com/en-us/training/modules/spatial-audio-tutorials-mrtk/5-exercise-spatialize-button-interaction-sounds

**Reply by imjusthere** [reply]
This is exactly what I was missing. Thank you!

**Comment by DebbyX3** [comment]
Hi, I was having the a similar issue using Microsoft Spatializer 2.0.37 on MRTK2.

I followed the whole learn guide (https://learn.microsoft.com/en-us/training/modules/spatial-audio-tutorials-mrtk) but I could only hear sounds when 'Spatialize' was unchecked in the Audio Source component.

At the end my problem was that the Master mixer did not have the effect "Microsoft Spatializer Mixer"

To solve:
- Create a new Audio Mixer (In the Project pane, right-click on the Assets folder, then Create > Audio Mixer) or use an already existing one
- Open the Audio Mixer pane: Window > Audio > Audio Mixer
- On the newly opened pane, select and open the mixer, and right-click, **then select Add effect at bottom > Microsoft Spatializer Mixer**

![immagine_2023-06-09_123421950](https://github.com/microsoft/MixedRealityToolkit-Unity/assets/26549164/b68b9af2-e275-45b3-b011-74a691eed0f4)


I was able to solve the issue by following these guides, especially the second one:
- https://github.com/MicrosoftDocs/mixed-reality/blob/docs/mixed-reality-docs/mr-dev-docs/develop/unity/spatial-sound-in-unity.md
- https://github.com/microsoft/spatialaudio-unity/blob/develop/Documentation/QuickStart.md#how-to-add-the-plugin-to-unity-projects
